uniflow/op/extract/load/utils.py

```diff
@@ -43,9 +47,18 @@ def read_file(source: str, mode: str = "r"):
 
         bucket_name, file_path = source[5:].split("/", 1)
         s3 = boto3.client("s3")
-        obj = s3.get_object(Bucket=bucket_name, Key=file_path)
-        data = obj["Body"].read()
-        return data.decode() if mode == "r" else data
+
+        file_type = os.path.splitext(file_path)[1][1:]
+
+        # TODO: Right now, our default for PDF Processing is to use local NougetModel, so we need to download the file to local and pass the path to the model
+        if file_type == "pdf":
+            file_name = os.path.join(LOCAL_FILE_PATH, file_path)
+            s3.download_file(Bucket=bucket_name, Key=file_path, Filename=file_name)
+            return {"pdf": file_name}
+        else:
+            obj = s3.get_object(Bucket=bucket_name, Key=file_path)
+            data = obj["Body"].read()
+            return data.decode() if mode == "r" else data
```

Comment:
qq: for html and txt, we perform I/O operation here. While for Nougat, I/O is postponed into the model_server.py NougatModelServer LazyDataset. Is it possible we can also perform I/O for pdf as well.

This should buy us two benefit.
- we have a unifed I/O process 
- I/O is all in extract, so we should be able to stream process extract and transform in the future.

Reply:
Ok. Let me see if I can modify the `NougatModelServer` logic. I will let you know